teamcity: failed tests on master: test/TestLogic

The following tests appear to have failed:

[#579972](https://teamcity.cockroachdb.com/viewLog.html?buildId=579972):


```
--- FAIL: test/TestLogic (156.590s)
test_log_scope.go:81: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic180049643
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: test/TestLogic: TestLogic/parallel-stmts (0.450s)

--- FAIL: test/TestLogic: TestLogic/parallel-stmts: TestLogic/parallel-stmts/rename_database (1.130s)
logic.go:2124: 
	testdata/logic_test/rename_database:41: error while processing
logic.go:2125: 
	testdata/logic_test/rename_database:41: 
	expected "relation \"kv\" does not exist", but found <nil>
------- Stdout: -------
=== PAUSE TestLogic/parallel-stmts/rename_database
--- FAIL: test/TestLogic (156.590s)
test_log_scope.go:81: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic180049643
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: test/TestLogic: TestLogic/parallel-stmts (0.450s)

--- FAIL: test/TestLogic: TestLogic/parallel-stmts: TestLogic/parallel-stmts/rename_database (1.130s)
logic.go:2124: 
	testdata/logic_test/rename_database:41: error while processing
logic.go:2125: 
	testdata/logic_test/rename_database:41: 
	expected "relation \"kv\" does not exist", but found <nil>
------- Stdout: -------
=== PAUSE TestLogic/parallel-stmts/rename_database

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/table.go
+++ b/pkg/sql/table.go
@@ -405,19 +405,24 @@ func (tc *TableCollection) releaseTables(ctx context.Context, opt releaseOpt) er
 			if !uc.dropped {
 				continue
 			}
+			// Wait until the database cache has been updated to properly
+			// reflect a dropped database, so that future commands on the
+			// same gateway node observe the dropped database.
 			err := tc.dbCacheSubscriber.waitForCacheState(
 				func(dc *databaseCache) (bool, error) {
-					desc, err := dc.getCachedDatabaseDesc(uc.name, false /*required*/)
-					if err != nil {
-						return false, err
+					// Resolve the database name from the database cache.
+					dbID, err := dc.getDatabaseID(ctx,
+						tc.leaseMgr.execCfg.DB.Txn, uc.name, false /*required*/)
+					if err != nil || dbID == 0 {
+						// dbID can still be 0 if required is false and
+						// the database is not found.
+						return true, err
 					}
-					if desc == nil {
-						return true, nil
-					}
+
 					// If the database name still exists but it now references another
-					// db, we're good - it means that the database name has been reused
-					// within the same transaction.
-					return desc.ID != uc.id, nil
+					// db with a more recent id, we're good - it means that the database
+					// name has been reused.
+					return dbID > uc.id, nil
 				})
 			if err != nil {
 				return err

